### PR TITLE
Add keep words token filter docs #8064

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
Add keep words token filter docs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8064

### Version
All

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AntonEliatra** [comment]
In this case the actual filter "type" is "keep", not "keep_words"
"keep_words" is a parameter that can be passed in.

I thought about renaming this to "keep" token filter across the changes, but it looks strange, especially since it uses KeepWordFilter Class. Just thought I'm mention it - in case anyone wants to change this.

**Comment by vagimeli** [comment]
@udabhas Will you review this PR for technical accuracy? Thank you.

**Comment by vagimeli** [comment]
> @udabhas Will you review this PR for technical accuracy? Thank you.

@udabhas @varun-lodaya This is over a month old. We need tech review approval to move it forward in the documentation process. Please review this week or provide a peer who can review it. Thank you.

File: _analyzers/token-filters/keep-words.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+layout: default
+title: Keep words
+parent: Token filters
+nav_order: 190
+---
+
+# Keep words token filter
+
+The `keep_words` token filter is designed to include only certain words in the analysis process. This filter is useful if you have a large body of text but are only interested in certain keywords or terms.
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/keep-words.md:10]:
```suggestion
The `keep_words` token filter is designed to keep only certain words during the analysis process. This filter is useful if you have a large body of text but are only interested in certain keywords or terms.
```

File: _analyzers/token-filters/keep-words.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+## Parameters
+
+The `keep_words` token filter can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`keep_words` |  Required if `keep_words_path` is not configured | List of strings | List of words to be kept.
+`keep_words_path` | Required if `keep_words` is not configured | String | Path to a file containing list of words to be kept.
+`keep_words_case` | Optional | Boolean | Lowercase all words during comparison. Default is `false`.
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/keep-words.md:20]:
```suggestion
`keep_words_case` | Optional | Boolean | Whether to lowercase all words during comparison. Default is `false`.
```

File: _analyzers/token-filters/keep-words.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ 
+
+## Example
+
+The following example request creates a new index named `my_index` and configures an analyzer with `keep_words` filter:
```
Comment (kolchfa-aws) [inline _analyzers/token-filters/keep-words.md:25]:
```suggestion
The following example request creates a new index named `my_index` and configures an analyzer with a `keep_words` filter:
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @AntonEliatra!  A couple of suggestions.

File: _analyzers/token-filters/index.md [hunk 4]
```diff
@@ -31,7 +31,7 @@ Token filter | Underlying Lucene token filter|  Description
...
 `hyphenation_decompounder` | [HyphenationCompoundWordTokenFilter](https://lucene.apache.org/core/9_8_0/analysis/common/org/apache/lucene/analysis/compound/HyphenationCompoundWordTokenFilter.html) | Uses XML-based hyphenation patterns to find potential subwords in compound words and checks the subwords against the specified word list. The token output contains only the subwords found in the word list.
 `keep_types` | [TypeTokenFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/core/TypeTokenFilter.html) | Keeps or removes tokens of a specific type.
-`keep_word` | [KeepWordFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/KeepWordFilter.html) | Checks the tokens against the specified word list and keeps only those that are in the list.
+[`keep_words`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/keep-words/) | [KeepWordFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/KeepWordFilter.html) | Checks the tokens against the specified word list and keeps only those that are in the list.
```
Comment (natebower) [inline _analyzers/token-filters/index.md:34]:
```suggestion
[`keep_words`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/keep-words/) | [KeepWordFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/KeepWordFilter.html) | Checks the tokens against the specified word list and keeps only those contained in the list.
```

File: _analyzers/token-filters/keep-words.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+The `keep_words` token filter is designed to keep only certain words during the analysis process. This filter is useful if you have a large body of text but are only interested in certain keywords or terms.
+
+## Parameters
+
+The `keep_words` token filter can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`keep_words` |  Required if `keep_words_path` is not configured | List of strings | List of words to be kept.
```
Comment (natebower) [inline _analyzers/token-filters/keep-words.md:18]:
```suggestion
`keep_words` |  Required if `keep_words_path` is not configured | List of strings | The list of words to keep.
```

File: _analyzers/token-filters/keep-words.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+`keep_words_path` | Required if `keep_words` is not configured | String | Path to a file containing list of words to be kept.
```
Comment (natebower) [inline _analyzers/token-filters/keep-words.md:19]:
```suggestion
`keep_words_path` | Required if `keep_words` is not configured | String | The path to the file containing the list of words to keep.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my changes and let me know if you have any questions. Thanks!

**Pull request merged**